lightning/src/ln/peer_handler.rs

```diff
@@ -2467,7 +2466,8 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, OM: Deref, L: Deref, CM
 			Some(peer_lock) => {
 				let peer = peer_lock.lock().unwrap();
 				if let Some((node_id, _)) = peer.their_node_id {
-					log_trace!(WithContext::from(&self.logger, Some(node_id), None, None), "Handling disconnection of peer {}", log_pubkey!(node_id));
+					let logger = WithContext::from(&self.logger, Some(node_id), None, None);
+					log_trace!(logger, "Handling disconnection of peer {} because {}", log_pubkey!(node_id), reason);
```

Comment:
nit: Looks like we include the node_id in the logger context and the log itself, which is a bit redundant